### PR TITLE
HDDS-5366.  [Ozone-Streaming] Implement stream method to ContainerStateMachine.

### PR DESCRIPTION
**Pull request by captainzmc**
## What changes were proposed in this pull request?

Use RATIS-Streaming to make ContainerStateMachine support writing streaming to the local.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-5366

## How was this patch tested?

CI test will be added after the client is finished

### CODE REVIEW

File: hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/transport/server/ratis/ContainerStateMachine.java [hunk 1]
```diff
@@ -501,6 +503,30 @@ private ExecutorService getCommandExecutor(
...
+      ContainerCommandResponseProto response = runCommand(
+          requestProto, context);
+      String path = response.getMessage();
+      CompletableFuture<DataStream> dataStream = new CompletableFuture<>();
+      dataStream.complete(new LocalStream(new StreamDataChannel(
+          Paths.get(path))));
+      return dataStream;
+    } catch (IOException e) {
+      throw new CompletionException("Failed to create data stream", e);
+    }
```
Comment (szetszwo) [inline hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/transport/server/ratis/ContainerStateMachine.java:527]: We should make it async like below:
```
  public CompletableFuture<DataStream> stream(RaftClientRequest request) {
    return CompletableFuture.supplyAsync(() -> {
      try {
        ContainerCommandRequestProto requestProto =
            getContainerCommandRequestProto(gid,
                request.getMessage().getContent());
        DispatcherContext context =
            new DispatcherContext.Builder()
                .setStage(DispatcherContext.WriteChunkStage.WRITE_DATA)
                .setContainer2BCSIDMap(container2BCSIDMap)
                .build();

        ContainerCommandResponseProto response = runCommand(
            requestProto, context);
        String path = response.getMessage();
        return new LocalStream(new StreamDataChannel(Paths.get(path))));
      } catch (IOException e) {
        throw new CompletionException("Failed to create data stream", e);
      }
    }, executor);
  }
```

**Review by szetszwo — COMMENTED** [review]
@captainzmc , thanks a lot for working on this.  Just have a minor comment.

BTW, let me update the HDDS-4454 branch with master.  Please update your branch as well.

**Review by szetszwo — APPROVED** [review]
+1 the change looks good.

**Comment by captainzmc** [comment]
Thanks @szetszwo  for the review.

**Pull request merged**